### ISSUE TITLE
test that RRSIG signature expiration / validation fields are respected

### ISSUE THREAD

**Issue by japaric** [hickory-dns/hickory-dns#2275]
this is a regression test for #2209

currently there's no API to tamper with a *signed* zone file and produce a *valid* signed zone file because that would require re-signing (regenerating the RRSIG records, etc.) the zone file and there's no code in `dns-test` that does that.

so, the easiest way to produce the bogus records required for this test could be to run the command `date --set` inside the container to fake the current time that the `ldns-signzone` tool sees.

note that these cases need to be tested:

- `inception > current_time` (signature produced in the future)
- `current_time > expiration` (signature has expired)
- `inception > expiration` (malformed record) would also be nice to have but probably `date --set` won't be sufficient to produce that kind of record

after #2253 lands these tests should become green given that a fix already landed in #2213

**Comment by justahero** [comment]
I think the `inception > expiration` test cannot be tested, because either the `inception` or the `expiration` timestamp are covered by the first two conditions.
